go/vt/vttablet/tabletmanager/restore.go

```diff
@@ -283,8 +283,9 @@ func (tm *TabletManager) restoreDataLocked(ctx context.Context, logger logutil.L
 		// Do nothing here, let the rest of code run
 		params.Logger.Infof("Dry run. No changes made")
 	default:
+		bgCtx := context.Background()
 		// If anything failed, we should reset the original tablet type
-		if err := tm.tmState.ChangeTabletType(ctx, originalType, DBActionNone); err != nil {
+		if err := tm.tmState.ChangeTabletType(bgCtx, originalType, DBActionNone); err != nil {
```

Comment:
I don't understand this change here. It looks like the opposite of the bugfix. Why would we not reuse `ctx`?

Reply:
We need to set status back to 'REPLICA' (or whatever prior to backup/restore) in all cases. If we use 'ctx' then incase of it being canceled or timed out we won't able to set the status back to original value (https://github.com/vitessio/vitess/issues/12701)